org.eclipse.ua.tests.doc/src/org/eclipse/ua/tests/doc/internal/linkchecker/ApiDocTest.java

```diff
@@ -276,31 +277,17 @@ private static InputSource getExtensionPointSchemaSource(IExtensionPoint extensi
 	private static void checkPackages(Set<String> packageIds, StringBuilder problems) {
 		Set<String> exportedPackageIds = new TreeSet<>();
 
-		exportedPackageIds.add("org.eclipse.core.runtime.adaptor"); // not exported, but makes sense to document since accessible from outside of OSGi framework
-		exportedPackageIds.add("org.eclipse.swt.ole.win32"); // somehow missing from State#getExportedPackages(), maybe because it's declared in the fragment only
+		exportedPackageIds.add("org.eclipse.core.runtime.adaptor"); // not exported, but makes sense to document since
+																	// accessible from outside of OSGi framework
+		exportedPackageIds.add("org.eclipse.swt.ole.win32"); // somehow missing from State#getExportedPackages(), maybe
+																// because it's declared in the fragment only
 		BundleContext context = FrameworkUtil.getBundle(ApiDocTest.class).getBundleContext();
 		ServiceReference<PlatformAdmin> platformAdminReference = context.getServiceReference(PlatformAdmin.class);
 		PlatformAdmin service = context.getService(platformAdminReference);
 		State state = service.getState(false);
-		ExportPackageDescription[] exportedPackages = state.getExportedPackages();
-		for (ExportPackageDescription exportPackageDescription : exportedPackages) {
-			String name = exportPackageDescription.getName();
-			if (!NON_API_PACKAGES.matcher(name).matches()) {
-				if (Boolean.TRUE.equals(exportPackageDescription.getDirective("x-internal"))
-						|| exportPackageDescription.getDirective("x-friends") != null) {
-					continue;
-				}
-				exportedPackageIds.add(name);
-//				Enumeration<String> packageChildren = exportPackageDescription.getSupplier().getBundle().getEntryPaths(name.replace('.', '/'));
-//				while (packageChildren != null && packageChildren.hasMoreElements()) {
-//					String child = packageChildren.nextElement();
-//					if (child.endsWith(".class")) {
-//						exportedPackageIds.add(name);
-//						break pack;
-//					}
-//				}
-//				System.out.append("Exported package without class files: ").append(name).append('\n');
-			}
+		for (BundleDescription bundle : state.getBundles()) {
+			bundle.getCapabilities(BundleRevision.PACKAGE_NAMESPACE)
+					.forEach((t) -> exportedPackageIds.add((String) t.getAttributes().get("osgi.wiring.package")));
```

Comment:
"osgi.wiring.package" is actually the value of BundleRevision.PACKAGE_NAMESPACE.

Reply:
Thanks, will try to remember it for the next time I play with wiring.